### PR TITLE
Get rid of magic numbers

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: src/terminator/mod.rs [hunk 1]
```diff
@@ -853,12 +855,50 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
...
-                let result = match name {
-                    30 => PrimVal::Bytes(4096), // _SC_PAGESIZE
-                    70 => PrimVal::from_i128(-1), // _SC_GETPW_R_SIZE_MAX
-                    _ => return Err(EvalError::Unimplemented(format!("Unimplemented sysconf name: {}", name)))
-                };
-                self.write_primval(dest, result, dest_ty)?;
+                // cache the sysconf integers
+                let paths = &[
+                    (&["libc", "_SC_PAGESIZE"], PrimVal::Bytes(4096)),
+                    (&["libc", "_SC_GETPW_R_SIZE_MAX"], PrimVal::from_i128(-1)),
```
Comment (eddyb) [inline src/terminator/mod.rs]: Whoa, this is a cool trick.

File: src/terminator/mod.rs [hunk 2]
```diff
@@ -853,12 +855,50 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
...
+                                trace!("pushing stack frame for global: {}", name);
+                                let frame = self.stack.len();
+                                self.push_stack_frame(
+                                    instance,
+                                    mir.span,
+                                    mir,
+                                    Lvalue::Global(cid),
+                                    cleanup,
+                                )?;
+                                while self.stack.len() != frame {
```
Comment (oli-obk) [inline src/terminator/mod.rs]: alternatively we could just return without modifying the current statement counter, so regular evaluation proceeds and this statement is reached automatically after the stackframe returns.

Comment (RalfJung) [inline src/terminator/mod.rs]: Any reason this is not using the new "evaluate to integer" thing? I actually thought this is what that PR was targeting. Either way this looks like it should get its own helper method.

File: src/terminator/mod.rs (hunk 1 above)
Comment (RalfJung) [inline src/terminator/mod.rs]: So where's the caching happening here? There seems to be no mutable state in this method; is it in the ectx? The comment should say that. ;)

Also, 4096 may not actually be the page size of this platform... do we have any way to get the right page size?

File: src/terminator/mod.rs (hunk 2 above)
Comment (eddyb) [inline src/terminator/mod.rs]: That PR was meant for use from `rustc_const_eval` I think.

File: src/terminator/mod.rs (hunk 1 above)
Comment (oli-obk) [inline src/terminator/mod.rs]: Yea I'll add a comment.

Uhm... can we just define 4096 as the page size of miri? idk, this is funky business anyway. I don't have a strong opinion.

File: src/terminator/mod.rs (hunk 2 above)
Comment (oli-obk) [inline src/terminator/mod.rs]: I'll create a helper method that'll get used both here and in the rustc int eval

File: src/terminator/mod.rs (hunk 1 above)
Comment (RalfJung) [inline src/terminator/mod.rs]: *shrug* fine for me.

**Pull request merged**